lib/mayaUsd/ufe/UsdTransform3dSetObjectMatrix.h

```diff
@@ -0,0 +1,95 @@
+//
+// Copyright 2021 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#pragma once
+
+#include <mayaUsd/ufe/UfeVersionCompat.h>
+#include <mayaUsd/ufe/UsdTransform3dBase.h>
+
+#include <pxr/base/gf/matrix4d.h>
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+namespace MAYAUSD_NS_DEF {
+namespace ufe {
+
+//! \brief Interface to set the matrix of objects in 3D.
+//
+// The UsdTransform3dMatrixOp and UsdTransform3dFallbackMayaXformStack classes
+// implement the Transform3d interface for a single matrix transform op and
+// for a fallback Maya transform stack, respectively.
+//
+// In both cases the complete transform stack will have additional transform
+// ops that define the complete 3D transformation for the whole object, while
+// only part of the complete transform stack (a single matrix and the fallback
+// Maya transform stack) is editable.
+//
+// Certain Maya operations (such as parent -absolute) require the capability to
+// edit the matrix transform of the whole object.  This class wraps a
+// UsdTransform3dMatrixOp or UsdTransform3dFallbackMayaXformStack and converts
```

Comment:
For UsdTransform3dMatrixOp targeting a single matrix op in a multi-matrix stack, and unconditionally for UsdTransform3dFallbackMayaXformStack, use this new UsdTransform3dSetObjectMatrix object to set the local transform of the object (e.g. for parent -absolute).

Reply:
You wrote a lot of great comments and documentation...but it still took me some time to figure out what exactly maps to Ml, Mw, Mr. Now looking at it back, what would help me to understand the code quicker is an example of a xform stack and how you break it down into these three transformations given a specific op we want to edit. It's all in the code, but having it described in one place (including pull request description) would help a lot.